pyproject.toml

```diff
@@ -167,6 +167,7 @@ exclude = [
 	"newsfragments*",
 	"docs",
 	"docs.*",
+	"build*",
```

Comment:
Somehow I found pesky `build/lib/**/*.py` files being included in the wheel if the `build` directory was not empty.

This is also the reason why I changed `MANIFEST.in`.

Reply:
Ugh. That doesn't seem right... adding a concern that seemingly affects all builds for something that's only affected when a bootstrap was run. Perhaps instead the bootstrap routine could remove the build directory after running?

Reply:
I think it would make more sense if the routine could remove the build directory before running...

I noted that if `build` exists `find_packages` will include it. Once it is listed in `packages`, changing `MANIFEST.in` does not seem to make an effect.

Reply:
Would it make more sense if instead we do:

```toml
[tool.setuptools.packages.find]
include = [
  "setuptools*",
  "pkg_resources*",
  "_distutils_hack*",
]
exclude = [
	"*.tests",
	"*.tests.*",
]
```

I think this simplifies things a bit and it is optimised for the case when the probability of adding `.py` in places outside the intended package directories is more likely than adding new package directories.

If the probabilities are reversed, then it makes less sense.

Reply:
> Ugh. That doesn't seem right... adding a concern that seemingly affects all builds for something that's only affected when a bootstrap was run. Perhaps instead the bootstrap routine could remove the build directory after running?

Probably the reason why `python -m build` works on the CI is because the job runs from a fresh clone (so the `build` directory is not there).

Reply:
I created a reproducer for the observed problem in https://github.com/pypa/setuptools/issues/4394.

(other than `pip install .` or `pip wheel .`, running the test suite with `tox` itself also produces the `build` directory, so if anyone tries to run `tox` before `pip install .`, they also end-up with spurious files).